Documentation/RepoCompose.md

```diff
@@ -0,0 +1,289 @@
+# Composing our repos
+
+In order to have automated builds that compose the output of our ever growing number of repos we need to get more structured data about their relationships.  Today repos provide nothing in the way of programmatic access to dependencies.  All of the information is known by the owners and as a result the builds and composition is fully manual. 
+
+Going forward repos will be given contracts, or APIs, that provide structured data about their dependencies and outputs.  This will give us the necessary data to compose our repos in our growing set of scenarios:
+
+- Linux distro builds from source
+- Official Windows signed build
+- Builds to quickly test and deploy new packages across repos
+
+These contracts will be included in the `run` command [specification](https://github.com/dotnet/buildtools/blob/master/Documentation/RunCommand.md).  This document in particular will be discussing the commands which allow us to compose the outputs of our repos.
+
+## Commands
+
+This document describes the commands used to compose different repos together.  There is a larger set of commands including `build`, `sync` and `clean` which is [described separately](https://github.com/dotnet/buildtools/blob/master/Documentation/Dev-workflow.md). 
+
+### consumes
+
+The `consumes` command returns json output that describes all of the external artifacts consumed by this repo.  This includes NuGet feeds, packages and arbitrary files from the web or file system.  The format of all these items is describe in the Artifact Specification section below. 
+
+The artifacts are grouped into sections: 
+
+- Build Dependencies: artifacts which are referenced by the build output and include NuGet packages, MSI, etc ...  In order to support a composed build these are further divided into floating and static dependencies:
```

Comment:
This information is more relevant to an orchestrator tool.  Essentially something like building up a build graph of the repos.  It would look at the consumes output, build dependencies, and combine that with the data in produces to determine build order.